### ISSUE TITLE
[android] Music Playback broken on Lollypop 5.1 (Api level 22)

### ISSUE THREAD

**Issue by gubatron** [frostwire/frostwire#905]
To replicate, try to play a song from Apollo's

```
05-21 18:04:05.804 4222-4810/com.frostwire.android I/MultiPlayer: setNextDataSource(): could not unset current player's next media playersetNextMediaPlayer failed.: status=0xFFFFFFED
     - invoked by  <MusicPlaybackService.java::setNextTrack:1433 on thread:MusicPlaybackService::MusicPlayerHandler(tid=469)>
    java.lang.IllegalArgumentException: setNextMediaPlayer failed.: status=0xFFFFFFED
        at android.media.MediaPlayer.setNextMediaPlayer(Native Method)
        at com.andrew.apollo.MusicPlaybackService$MultiPlayer.setNextDataSource(MusicPlaybackService.java:2833)
        at com.andrew.apollo.MusicPlaybackService.setNextTrack(MusicPlaybackService.java:1433)
        at com.andrew.apollo.MusicPlaybackService.setRepeatMode(MusicPlaybackService.java:2514)
        at com.andrew.apollo.MusicPlaybackService.initRepeatModeAndShuffleTask(MusicPlaybackService.java:533)
        at com.andrew.apollo.MusicPlaybackService.initService(MusicPlaybackService.java:877)
        at com.andrew.apollo.MusicPlaybackService.lambda$Qefg4qJy97KSKzGe3agIPPeITrY(MusicPlaybackService.java)
        at com.andrew.apollo.-$$Lambda$MusicPlaybackService$Qefg4qJy97KSKzGe3agIPPeITrY.run(lambda)
        at android.os.Handler.handleCallback(Handler.java:739)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:135)
        at android.os.HandlerThread.run(HandlerThread.java:61)
```

Other hints, when playing from "My Files > My Audio"

```
05-21 18:06:12.421 2251-4583/android.process.media W/MediaProvider: original media doesn't exist or it's canceled.
05-21 18:06:15.765 4222-4245/com.frostwire.android I/UIUtils: playEphemeralPlaylistTask() is MusicPlaybackService running? false
05-21 18:06:15.765 4222-4245/com.frostwire.android I/MusicUtils: startMusicPlaybackService() startService(MusicPlaybackService)
     - invoked by  <UIUtils.java::playEphemeralPlaylistTask:650 on thread:Engine-thread-7393-@295246211(tid=212)>
05-21 18:06:15.767 1541-1795/system_process W/AudioTrack: AUDIO_OUTPUT_FLAG_FAST denied by client
05-21 18:06:15.773 4222-4222/com.frostwire.android I/MusicPlaybackService: onStartCommand: Got new intent Intent { cmp=com.frostwire.android/com.andrew.apollo.MusicPlaybackService }, startId = 6
05-21 18:06:15.773 4222-4222/com.frostwire.android I/MusicPlaybackService: handleCommandIntent: action = null, command = null
05-21 18:06:15.773 4222-4222/com.frostwire.android I/MusicPlaybackService: handleCommandIntent: nothing to be done here, exiting.
```

Looks like MusicPlaybackService initialization is borked on 5.1.1, somehow. Unbelievable.

**Comment by gubatron** [comment]
update: we've got a warm trail

<img width="366" alt="Screen Shot 2020-05-21 at 6 15 30 PM" src="https://user-images.githubusercontent.com/163977/82618018-1305d780-9b8f-11ea-9bda-f9157edf400b.png">

Now it's time to dust off Android Manuals from 2015-2016 for Media Playback on Android Lollypop 5.1 and see what we broke.

**Comment by gubatron** [comment]
Fixed by 233a618ff6b20882db924a221643b8dee0245f27
